routes/dao_coin_exchange_with_fees.go

```diff
@@ -112,28 +134,24 @@ func (fes *APIServer) UpdateDaoCoinMarketFees(ww http.ResponseWriter, req *http.
 		return
 	}
 
-	// When this is nil then the UpdaterPublicKey is assumed to be the owner of
-	// the profile.
-	var profilePublicKeyBytes []byte
-	if requestData.ProfilePublicKeyBase58Check != "" {
-		profilePublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.ProfilePublicKeyBase58Check)
-		if err != nil || len(profilePublicKeyBytes) != btcec.PubKeyBytesLenCompressed {
-			_AddBadRequestError(ww, fmt.Sprintf(
-				"UpdateDaoCoinMarketFees: Problem decoding public key %s: %v",
-				requestData.ProfilePublicKeyBase58Check, err))
-			return
-		}
+	if requestData.ProfilePublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"UpdateDaoCoinMarketFees: ProfilePublicKeyBase58Check and UpdaterPublicKeyBase58Check must be provided"))
+		return
 	}
 
-	// Get the public key.
-	profilePublicKey := updaterPublicKeyBytes
-	if requestData.ProfilePublicKeyBase58Check != "" {
-		profilePublicKey = profilePublicKeyBytes
+	// Decode the profile public key.
+	profilePublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.ProfilePublicKeyBase58Check)
```

Comment:
You could remove some duplicated code here by using `GetPubKeyBytesFromBase58Check`